### DISCUSSION TITLE
Blog index page without except

### DISCUSSION THREAD

**Discussion by x4Cx58x54** [squidfunk/mkdocs-material#6210] [Ideas]
This is a feature request for the minimalist bloggers: adding a `none` option for `blog.post_excerpt` for those who want only their post titles show on the index page, which means removing the post excerpt and the "continue reading" link.

**Comment by alexvoss** [comment]
This should be possible to achieve with customization. You can simply hide the excerpt and the link below it by setting them to `display: none` in `extra.css`. Material for MkDocs is highly [customizable](https://squidfunk.github.io/mkdocs-material/customization) and it is not possible to add everything people might need as a selectable option.

**Reply by x4Cx58x54** [reply]
Thank you for answering. Yes I have tried hiding the excerpt using css. However the element of post titles and post contents in the excerpts are in the same html level, thus difficult to apply selectors to the contents in a neat and clean implementation.

**Reply by alexvoss** [reply]
How about this? It is probably not a full solution as you may want to reduce the whitespace but should get you started:

```css
article.md-post--excerpt p, article.md-post--excerpt nav {
  display: none;
}
```

**Comment by mfridman** [comment]
Ye, this would be a nice quality-of-life improvement (although I suspect you could do this with a custom template by category, I need to remember how that works). 

To add another data point, I have a blog split into sections (by category): Articles, Fragments, Snippets, etc. And for different categories, I'd like to have different behaviour.

For Articles, I'd like to display the blog excerpt, but for Snippets, I only want a title. Going down the .css route it's an all-or-nothing change.

I could add `<!-- more -->` but then it'll still have a new line with a "Continue reading".

<img src="https://github.com/squidfunk/mkdocs-material/assets/6278244/6d22df55-af53-4885-b196-d7517c216ab1" width=350px>

**Reply by kamilkrzyskow** [reply]
As the `in Snippets` category information is in the header of the excerpt and you want to remove the content of the excerpt, then I think CSS won't be able to handle such conditional. You could however use extra JavaScript to achieve that:
https://squidfunk.github.io/mkdocs-material/customization/#additional-javascript
```js
let removeInnerExcerpt = (excerpt) => {
  let contentElement = excerpt.querySelector(".md-post__content");
  for (let node of contentElement.childNodes) {
    if (node.tagName && node.tagName !== "H2") {
      // perhaps removing nodes from a live view isn't ideal
      // but the result is as intended
      node.remove();
    }
  }
}

document.addEventListener("DOMContentLoaded", _ => {
  let postExcerpts = document.querySelectorAll(".md-post--excerpt");
  if (!postExcerpts) {
    console.debug("no excerpts found");
    return;
  }

  // keep them lowercase
  let categoriesWithRemovedExcerpts = ["general", "search"];

  for (let excerpt of postExcerpts) {
    let metaLinks = excerpt.querySelectorAll(".md-meta__link");

    for (let link of metaLinks) {
      let category = link.innerText.trim().toLowerCase();
      if (categoriesWithRemovedExcerpts.includes(category)) {
        removeInnerExcerpt(excerpt);
        break;
      }
    }
  }
});
```
Just modify the `categoriesWithRemovedExcerpts` with the names of categories that would have their excerpts removed, leaving only the title. Tested on the official Material for MkDocs blog index.

**Reply by squidfunk** [reply]
Unfortunately, this might lead to content shift on slow connections, because the content is visible, until the JavaScript runs and it isn't. IMHO this should be done in the template, not in JavaScript 😉

**Reply by squidfunk** [reply]
@mfridman you can use a combination of [custom index pages](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#custom-index-pages) for your categories and a custom template.  For example, say you have the categories `foo` and `bar`, then your file layout could look like this:

``` sh
.
├─ overrides/
│  ├─ blog-category-foo.html
│  └─ blog-category-bar.html
├─ docs/
│  └─ blog/
│     ├─ category/
│     │  ├─ foo.md 
│     │  └─ bar.md 
│     ├─ posts/
│     └─ index.md
└─ mkdocs.yml
```

In `mkdocs.yml`:

``` yaml
theme:
  name: material
  custom_dir: overrides
```

In `foo.md`:

``` markdown
---
template: blog-category-foo.html
---

# Welcome to Foo!

Custom content

<!-- Blog posts will be appended here -->
```

Same goes for `bar.md`. The idea is that you override the [`blog.html`](https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/blog.html) in `blog-category-foo.html` (and subsequently `blog-post.html`, as that's the partial that is used for rendering the excerpts). I understand that this is a little more involved, but before adding additional rendering options, we'd like to learn about the common patterns that people use to build their blogs.

**Reply by mfridman** [reply]
Thanks for the example, that's exactly the route I was going down. I also think it's a fair tradeoff for users who want more customization.

Many suggestions (like this discussion) add complexity, and overhead, and are quite subjective. Seems like the 80/20 principle, where simpler solutions will generally suffice for most users. I like your approach of waiting and getting feedback.

**Reply by mfridman** [reply]
Fwiw this turned out nicely (screenshot is what I was going for), I might do a write-up in case anyone else runs into this (or my future self, hehe).

<div align=center>
<img src="https://github.com/squidfunk/mkdocs-material/assets/6278244/013263b6-9c59-47e1-acab-5401e8f287fc" width=550>
</div>

**Reply by squidfunk** [reply]
Love it!
